rs/messaging/src/routing/stream_handler/tests.rs

```diff
@@ -291,41 +279,27 @@ fn induct_loopback_stream_reroute_response() {
                 CANISTER_MIGRATION_SUBNET => migration_stream,
             ]);
 
-            let initial_available_guaranteed_response_memory =
-                stream_handler.available_guaranteed_response_memory(&state);
             let mut available_guaranteed_response_memory =
-                initial_available_guaranteed_response_memory;
+                stream_handler.available_guaranteed_response_memory(&state);
 
             let inducted_state = stream_handler
                 .induct_loopback_stream(state, &mut available_guaranteed_response_memory);
 
             assert_eq!(expected_state, inducted_state);
-
-            // One request inducted, one response inducted and one reject response produced.
+            // `available_guaranteed_response_memory` does not keep track of gc'ing
+            // the response @22 in the loopback stream after inducting it.
             assert_eq!(
-                initial_available_guaranteed_response_memory
-                    // Inducting a request triggers a new reservation in the output queue.
-                    - MAX_RESPONSE_COUNT_BYTES as i64
-                    // Inducting a response uses memory for the response, but also frees a reservation.
-                    - (inducted_response_count_bytes as i64 - MAX_RESPONSE_COUNT_BYTES as i64)
-                    // The reject response is in the loopback stream, i.e. no reservation is freed.
-                    - reject_response_count_bytes as i64,
-                available_guaranteed_response_memory,
-            );
-            // Not equal, because the computed available memory does not account for the
-            // reject response (since it's from a canister no longer hosted by the subnet).
-            assert!(
-                stream_handler.available_guaranteed_response_memory(&inducted_state)
-                    >= available_guaranteed_response_memory
+                available_guaranteed_response_memory + inducted_response_count_bytes as i64,
+                stream_handler.available_guaranteed_response_memory(&inducted_state),
```

Comment:
To be honest, I find the way I wrote this here both snappier and less confusing. It shows directly that the computed value and the tracked value differ by a few bytes because garbage collection is not tracked properly. It also required less thinking on my side to get it right.

Reply:
There are other places where I could rewrite it like this if people agree with my assessment.

Reply:
I suppose it's OK either way: as it was before, it was more verbose, but it actually looked for an absolute value (and it allowed the reader to track exactly how much memory was allocated for what); as it is now, it's more concise, but it's merely a relative comparison of the current actual usage vs. a pessimistic estimate (so you wouldn't find out from this assert alone if e.g. the implementation somehow forgot to account for the reject response).

Given the above, I would rather have a mix of absolute and relative comparisons than only relative comparisons everywhere.

Reply:
As I see it remodelling what `available_guaranteed_response_memory()` does here is out of place since there should be a test for that somewhere else. The new way more explicitly shows what the difference is in the context of what we are testing here without overloading this test with information about an otherwise unrelated function.